include/alpaka/onHost/algo/scan.hpp

```diff
@@ -0,0 +1,162 @@
+/* Copyright 2025 Anton Reinhard
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include "alpaka/api/trait.hpp"
+#include "alpaka/concepts.hpp"
+#include "alpaka/onHost/algo/internal/scan.hpp"
+
+namespace alpaka::onHost
+{
+    /** @brief For a scan of some size, this function returns the necessary buffer size.
+     *
+     * When multiple scans of the same extents are needed (for example in a loop), this function can be used to only
+     * allocate an intermediate buffer once, removing alloc/free overhead. For unique scan calls, the buffer can be
+     * omitted in the scan call, in which case it will be allocated and freed on the fly.
+     *
+     * @param extents The extents of the scan.
+     * @return The size of the buffer to allocate in number of elements. The buffer must have the same data type as the
+     * object being scanned.
+     */
+    auto getScanBufferSize(alpaka::concepts::Vector auto const& extents)
+    {
+        return internal::scanBufferSize(extents);
+    }
+
+    /** @brief Perform an inclusive scan on the input data and write the result to the output data.
+     *
+     * @tparam T_Data Data type that will be scanned. T_Data{0} should be the neutral element for addition.
+     * @param exec The executor to run with.
+     * @param devAcc The device to run on.
+     * @param queue The queue to enqueue to.
+     * @param inputVec The input data. Can be const.
+     * @param outputVec The output data. To perform an in-place scan, use the overload with only one data object.
+     * @param buffer (optional) The internally used buffer. Use the scanBufferSize() function to check how big the
+     * buffer needs to be. If omitted, it will be allocated and destructed on the fly.
+     */
+    template<typename T_Data>
+    void inclusiveScan(
+        alpaka::concepts::Executor auto& exec,
+        alpaka::onHost::concepts::Device auto& devAcc,
+        auto const& queue,
+        alpaka::concepts::IDataSource<T_Data> auto const& inputVec,
+        alpaka::concepts::IMdSpan<T_Data> auto outputVec,
+        auto... buffer)
+    {
```

Comment:
```suggestion
    void inclusiveScan(
        alpaka::concepts::Executor auto& exec,
        alpaka::onHost::concepts::Device auto& devAcc,
        auto const& queue,
        alpaka::concepts::IDataSource auto const& inputVec,
        alpaka::concepts::IMdSpan auto outputVec,
        auto... buffer)
        requires std::same_as<std::remove_cv<ALPAKA_TYPE_OF(inputVec)::value_type>, std::remove_cv<ALPAKA_TYPE_OF(outputVec)::value_type>>
    {
```

I understand, that you want to have the `T_Data` type to make the concepts more explicit. But this requires that you need to set the data type manually for each function, because we cannot deduce it in the function signature. I think it is better to check if buffer `value_types` are equal via required clause.

By the way, I was also thinking about to add a `raw_type` to `IDataSource` that you can easily compare the value type, without the need to remove a possible `const`. @psychocoderHPC What are you thinking about?

Reply:
I added static_asserts in the internal implementation